portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -95,14 +103,22 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) t
     query.addSort(sortOrder.property, sortOrder.sortDirection);
 
     // Create comment list
-    PreparedQuery results = datastore.prepare(query);
+    PreparedQuery preparedQuery = datastore.prepare(query);
 
+    QueryResultList<Entity> results;
+    try {
+      results = preparedQuery.asQueryResultList(fetchOptions);
+    } catch(IllegalArgumentException e) {
+      return new CommentsResult();
```

Comment:
Should you return an error instead?

e.g. `response.setStatus(HttpServletResponse.SC_BAD_REQUEST);`